src/estimation/alignment_properties.rs

```diff
@@ -13,18 +15,120 @@ use ordered_float::NotNan;
 use rust_htslib::bam::{self, record::Cigar};
 use statrs::statistics::{OrderStatistics, Statistics};
 
-#[derive(Clone, Debug, Copy, Deserialize, Serialize)]
+use crate::reference;
+use crate::utils::SimpleCounter;
+
+pub(crate) const MIN_HOMOPOLYMER_LEN: usize = 4;
+
+fn default_homopolymer_error_model() -> HashMap<i8, f64> {
+    let mut model = HashMap::new();
+    model.insert(0, 0.9975414130829068);
+    model.insert(1, 0.0010076175889726332);
+    model.insert(-1, 0.0010076175889726332);
+    model.insert(-2, 0.00020152351779452663);
+    model.insert(2, 0.00010076175889726332);
+    model.insert(3, 5.038087944863166e-05);
+    model.insert(-3, 9.068558300753699e-05);
+
+    model
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
 pub(crate) struct AlignmentProperties {
     pub(crate) insert_size: Option<InsertSize>,
     pub(crate) max_del_cigar_len: Option<u32>,
     pub(crate) max_ins_cigar_len: Option<u32>,
     pub(crate) frac_max_softclip: Option<f64>,
     pub(crate) max_read_len: u32,
+    #[serde(default = "default_homopolymer_error_model")]
+    pub(crate) homopolymer_error_model: HashMap<i8, f64>,
     #[serde(default)]
     initial: bool,
 }
 
 impl AlignmentProperties {
+    pub(crate) fn update_homopolymer_error_model(
```

Comment:
@tedil can you please have a close look at this and check whether the error ratest estimated here would make sense as parameters for the homopolypair-hmm?

Reply:
So the current homopolypair-hmm implementation assumes that extending a homopolymer (e.g. from 2,1 to 3,1 to 4,1 etc) always happens at **roughly** the same rate (which is what I observed in some preliminary experiments on GIAB data) [there's only one homopolymer state for each base {A/T homopolymers are more likely than C/G ones if I recall correctly} and for target/query, not multiple sequential states, though we could easily change that I guess].
i.e. there's one initial homopolymer-open probability akin to gap-open and one homopolymer-extend probability akin to gap-extend.

From what I recall, the extend probability is, depending on base+target/query, around 0.04 to 0.18 or so, so at least the 1 → 2 step is fine. Not sure about the initial probabilities atm. Will report back later.

Reply:
Good point. It would be fine for me if you extended the homopolymer error model estimation in the alignment properties to be base specific if the rest fits.